SPDY/SPDYCacheStoragePolicy.m

```diff
@@ -0,0 +1,79 @@
+//
+//  SPDYCacheStoragePolicy.m
+//  SPDY
+//
+//  Copyright (c) 2014 Twitter, Inc. All rights reserved.
+//  Licensed under the Apache License v2.0
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Derived from code in Apple, Inc.'s CustomHTTPProtocol sample
+//  project, found as of this notice at
+//  https://developer.apple.com/LIBRARY/IOS/samplecode/CustomHTTPProtocol
+//
+
+#import "SPDYCacheStoragePolicy.h"
+
+extern NSURLCacheStoragePolicy SPDYCacheStoragePolicy(NSURLRequest *request, NSHTTPURLResponse *response)
+{
+    bool                     cacheable;
+    NSURLCacheStoragePolicy  result;
+
+    // First determine if the request is cacheable based on its status code.
+    
+    switch (response.statusCode) {
+        case 200:
+        case 203:
+        case 206:
+        case 301:
+        case 304:
+        case 404:
+        case 410:
+            cacheable = YES;
+            break;
+        default:
+            cacheable = NO;
+            break;
+    }
+
+    // If the response might be cacheable, look at the "Cache-Control" header in 
+    // the response.
+
+    // IMPORTANT: We can't rely on -rangeOfString: returning valid results if the target 
+    // string is nil, so we have to explicitly test for nil in the following two cases.
+
+    if (cacheable) {
+        NSString *responseHeader;
+
+        responseHeader = [response.allHeaderFields[@"cache-control"] lowercaseString];
```

Comment:
how do you know the header field will be lower case?  In my experience Apple has a set of fields that it automatically coerces to be camel case (which could change over time).

I think it might be better to enumerate the fields at the top in one loop and pull out all the values of interest by matching on case-insensitive comparison.